### ISSUE TITLE
Prove cardinal shoe division without choice

### ISSUE THREAD

**Issue by jkingdon** [metamath/set.mm#3505]
That is, prove that `( N e. _om /\ (/) e. N /\ ( A X. N ) ~~ ( B X. N ) ) -> A ~~ B` in set.mm without using choice. The term "shoe division" refers to the ordered nature of a natural number and is after Russell's sock/shoe example in explaining the axiom of choice. (see especially the Lutz paper cited below).

If this is already in set.mm, speak up, but I didn't see any relevant search results for `( ? X. ? ) ~~ ( ? X. ? )`

References:
* 2023 discussion thread: https://mathstodon.xyz/@AndresCaicedo/111106716181993036
* "Conway and Doyle Can Divide by Three, But I Can't" by Patrick Lutz, 2023, https://arxiv.org/abs/2309.11634
* "On Dividing by Two in Constructive Mathematics", Andrew Swan, 2018, https://arxiv.org/abs/1804.04490

**Comment by tirix** [comment]
Yes, I also don't see anything similar in set.mm yet. That would be a nice addition!
